aiida/manage/tests/pytest_fixtures.py

```diff
@@ -521,6 +525,155 @@ def get_code(entry_point, executable, computer=aiida_localhost, label=None, prep
     return get_code
 
 
+@pytest.fixture(scope='session')
+def generate_ssh_key() -> t.Generator[pathlib.Path, None, None]:
+    """Generate a temporary SSH key pair for the test session and add it to the ``.ssh`` folder of the current user."""
+    from cryptography.hazmat.backends import default_backend as crypto_default_backend
+    from cryptography.hazmat.primitives import serialization as crypto_serialization
+    from cryptography.hazmat.primitives.asymmetric import rsa
+
+    key = rsa.generate_private_key(
+        backend=crypto_default_backend(),
+        public_exponent=65537,
+        key_size=2048,
+    )
+
+    private_key = key.private_bytes(
+        crypto_serialization.Encoding.PEM,
+        crypto_serialization.PrivateFormat.PKCS8,
+        crypto_serialization.NoEncryption(),
+    )
+
+    public_key = key.public_key().public_bytes(
+        crypto_serialization.Encoding.OpenSSH,
+        crypto_serialization.PublicFormat.OpenSSH,
+    )
+
+    dirpath = pathlib.Path('~').expanduser() / '.ssh'
```

Comment:
Do we have to put this in the `~/.ssh` directory, which may be used for other purposes during the test?

Would it be possible to place the key in a temporary directory instead?

Reply:
To be honest, I just added it there because I thought that was maybe required for it to work with paramiko out of the box. But maybe if we pass the explicit filepath anyway, it could be anywhere. Will use a temp directory and see if the tests pass.